### PR TITLE
[CI] Fix XPU CI failure

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138548

# Motivation
Fix https://github.com/pytorch/pytorch/issues/138577.

# Solution
1. All UTs in `test/inductor/test_compiled_optimizers.py` are fixed by https://github.com/pytorch/pytorch/pull/134170
2. UT in `test/inductor/test_pattern_matcher.py` is introduced by https://github.com/pytorch/pytorch/pull/138089, we will skip this UT due to the unsupported feature `max_autotune_gemm_backends:Triton`.
3. We have a new impl related to `histc`, so we remove the expected failure from `test/inductor/test_torchinductor_opinfo.py`
4. We support `avg_pool3d` for `fp16` data type, so we remove the expected failure from `test/inductor/test_torchinductor_opinfo.py`
5. CUDA-bias code is introduced by https://github.com/pytorch/pytorch/issues/138472, we just generalize it to `GPU_TYPE`.

# Additional Context
> Why update torch-xpu-ops commit pin here?

We have to update commit pin to avoid the build failure raised by the code change [C10_UNUSED](https://github.com/pytorch/pytorch/pull/138364).

> What does the feature of torch-xpu-ops update?

1. Add some foreach ops, like `unary ops` and `foreach_clamp_max` etc;
2. Add some maxpool ops forward and backward, like `averge_pool3d` and `max_pool3d`
3. Add some other ops, like `log_normal_`, `index_copy`, and `mode` etc;
4. fix build failure related to `C10_UNUSED`;

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: test/inductor/test_torchinductor_opinfo.py [hunk 1]
```diff
@@ -351,7 +351,6 @@ def format_op(op):
     "nn.functional.conv_transpose3d": {f32, f64},
     # rrelu not supported on XPU now
     "nn.functional.rrelu": {f16, f32, f64},
-    "histc": {i32, i64},
```
Comment (guangyey) [inline test/inductor/test_torchinductor_opinfo.py:354]: This is fixed in torch-xpu-ops. We have a histc impl now.

File: test/inductor/test_compiled_optimizers.py [hunk 2]
```diff
@@ -167,14 +167,11 @@ class KernelCounts(NamedTuple):
...
     "test_nadam_tensor_lr_weight_decay_momentum_decay_decoupled_weight_decay_capturable_cuda": 6,
     "test_nadam_tensor_lr_weight_decay_momentum_decay_decoupled_weight_decay_capturable_xpu": 9,
-    "test_nadam_tensor_lr_weight_decay_momentum_decay_decoupled_weight_decay_capturable_foreach_xpu": 3,
     "test_radam_tensor_lr_capturable_weight_decay_decoupled_weight_decay_cuda": 6,
     "test_radam_tensor_lr_capturable_weight_decay_decoupled_weight_decay_xpu": 6,
-    "test_radam_tensor_lr_capturable_weight_decay_decoupled_weight_decay_foreach_xpu": 3,
     "test_sgd_tensor_lr_cpu": 2,
     "test_sgd_tensor_lr_cuda": 2,
     "test_sgd_tensor_lr_xpu": 2,
-    "test_sgd_tensor_lr_foreach_xpu": 2,
```
Comment (guangyey) [inline test/inductor/test_compiled_optimizers.py:177]: This is fixed by https://github.com/pytorch/pytorch/pull/134170

File: test/inductor/test_torchinductor_opinfo.py [hunk 3]
```diff
@@ -351,11 +351,9 @@ def format_op(op):
...
     # not implemented for 'Half'
     "nn.functional.multilabel_margin_loss": {f16},
     "nn.functional.multi_margin_loss": {f16},
-    "nn.functional.avg_pool3d": {f16},
```
Comment (guangyey) [inline test/inductor/test_torchinductor_opinfo.py:358]: `fp16` data type is supported on avg_pool3d for xpu ops.

**Review by malfet — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

**Comment by guangyey** [comment]
@malfet Update torch-xpu-ops pin commit from a developing branch to a master branch—no more other code changes. Please help review again.

**Review by malfet — APPROVED** [review]
Nice hashsum!

**Comment by guangyey** [comment]
@malfet Due to the newly updated pin commit introducing a sort of new op implementation, the expected failure UTs in the Inductor will be fixed. To avoid introducing new code changes in this PR, we would change back to the previously updated pin commit and let this PR land first. Additionally, we will file another PR to remove these expected failures with a new torch-xpu-ops pin commit.

**Pull request closed without merging**